### PR TITLE
[Backport] Fix SKU limit in import new products

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fix SKU limit in import new products
### Fixed Issues (if relevant)

1. magento/magento2#17865: import new products via csv: products are created with empty value when strings are too long

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

Backport for: https://github.com/magento/magento2/pull/18639

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18591) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/CatalogImportExport/Model/Import/Product.php [hunk 1]
```diff
@@ -159,6 +159,11 @@ class Product extends \Magento\ImportExport\Model\Import\Entity\AbstractEntity
      */
     const URL_KEY = 'url_key';
 
+    /**
+    * SKU max length
+    */
+    const DB_MAX_SKU_LENGTH = 65;
```
Comment (slavvka) [inline app/code/Magento/CatalogImportExport/Model/Import/Product.php]: You introduced the constant in the class marked with `@api` that is backward incompatible (https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/). Additionally the size of `sku` column is 64, not 65. Please reimplement the solution.

**Review by slavvka — CHANGES_REQUESTED** [review]
Thank you for the contribution! Please make changes according to the review results.

**Comment by slavvka** [comment]
Also we want the fixes to be delivered to 2.3 and then backported to 2.2. Please create PR to 2.3

File: app/code/Magento/CatalogImportExport/Model/Import/Product.php (hunk 1 above)
Comment (ravi-chandra3197) [reply app/code/Magento/CatalogImportExport/Model/Import/Product.php]: > Also we want the fixes to be delivered to 2.3 and then backported to 2.2. Please create PR to 2.3
Hello, I have created a new pull request #18639 as per your suggestion based on 2.3 develop branch. Please kindly review it.

Comment (slavvka) [reply app/code/Magento/CatalogImportExport/Model/Import/Product.php]: Done. Thank you for urgent reaction :)

**Comment by ravi-chandra3197** [comment]
Hi @slavvka can you review it
I have changed as per your suggestions now it allows the product with SKU that contains 64 characters.

**Comment by ravi-chandra3197** [comment]
Hi @slavvka can you review it
I have changed as per your suggestions now it allows the product with SKU that contains 64 characters.

**Comment by ravi-chandra3197** [comment]
Hi @slavvka can you review it
I have changed as per your suggestions now it allows the product with SKU that contains 64 characters.

**Comment by ravi-chandra3197** [comment]
@slavvka can you review it
I have changed as per your suggestions now it allows the product with SKU that contains 64 characters.

**Comment by ravi-chandra3197** [comment]
@slavvka can you review it
I have changed as per your suggestions now it allows the product with SKU that contains 64 characters.

**Comment by slavvka** [comment]
hey @ravi-chandra3197 thank you for the changes. Looks good

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-3321](https://jira.corp.magento.com/browse/ENGCOM-3321) has been created to process this Pull Request

**Review by slavvka — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @slavvka, thank you for the review.
[ENGCOM-3321](https://jira.corp.magento.com/browse/ENGCOM-3321) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.8. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**